independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/buildextension/beans/SyntheticInjectionPointInstanceTest.java

```diff
@@ -0,0 +1,84 @@
+package io.quarkus.arc.test.buildextension.beans;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.util.List;
+
+import jakarta.enterprise.inject.Instance;
+import jakarta.enterprise.util.TypeLiteral;
+import jakarta.inject.Singleton;
+
+import org.jboss.jandex.ClassType;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.ParameterizedType;
+import org.jboss.jandex.Type;
+import org.jboss.jandex.Type.Kind;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.BeanCreator;
+import io.quarkus.arc.processor.BeanRegistrar;
+import io.quarkus.arc.test.ArcTestContainer;
+
+public class SyntheticInjectionPointInstanceTest {
+
+    @RegisterExtension
+    public ArcTestContainer container = ArcTestContainer.builder()
+            .beanClasses(SomethingRemovable.class)
+            .removeUnusedBeans(true)
+            .beanRegistrars(new TestRegistrar()).build();
+
+    @SuppressWarnings("serial")
+    @Test
+    public void testBeanNotRemoved() {
+        List<String> list = Arc.container().instance(new TypeLiteral<List<String>>() {
+        }).get();
+        assertNotNull(list);
+        assertEquals(1, list.size());
+        assertEquals(SomethingRemovable.STR, list.get(0));
```

Comment:
Maybe also check that `SomethingRemovable.class`  can be looked up independently?

Reply:
Well, this tests that no other `@Inject SomethingRemovable` or `@Inject Instance<SomethingRemovable>` makes the `SomethingRemovable` bean unremovable. And `CDI.current()` and friends use the same logic as the `Instance` synthetic injection point.

Reply:
Fair enough.